Explore Variant_Classification's Impact on TMB: Part 2: Definition discrepancies - how often do they come up?

cansavvy opened the pull request:
### Purpose/implementation Section

⚠️  ***This PR includes changes filed in #739; that should be reviewed first***

#### What scientific question is your analysis addressing?
This address part 2 of #729 

> 2) How often do these: `Translation_Start_Site`, `Nonstop_Mutation` or `Splice_Site` come up? (These are not included in the FoCR definitions, but are included in `maftools` definition. 

There's not very many mutations that are affected and given the results of part 1 (#739) I do not consider this to be something that should be a big concern moving forward. 

#### What was your approach?
Pretty straightforward: 
- Import consensus mutation files for TCGA and PBTA. 
- Make barplots of the `Variant_Classification` variable. 
- Label which are included in the FoCR nonsynonymous definition and which are only noted by `maftools` nonsynonymous definition. 

I also added an option to `calculate_tmb.R` script that allows us to choose which nonsynonymous mutation definition we want to use for filtering. So now there's `--nonsynfilter_maf` or `--nonsynfilter_focr`. 

#### What GitHub issue does your pull request address?
#729 

### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.
- Note that this uses the consensus file which is based on strelka, mutect, and Lancet and does NOT have coding region filtering done. So it doesn't exactly answer about TMB directly, but is related. 
- Does this address the scientific questions at hand? 
- Are there further analyses we'd like to see here? 

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes

### Results

#### What types of results are included (e.g., table, figure)?

Here's the rendered notebook: 
[explore_var_class_discrepancies.nb.html.zip](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/5109618/explore_var_class_discrepancies.nb.html.zip)

#### Reproducibility Checklist

<!-- Check all those that apply or remove this section if it is not applicable.-->

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

<!-- Please review https://github.com/AlexsLemonade/OpenPBTA-analysis#documenting-your-analysis -->

I haven't added a README since this is a side analysis. The main information is in the Rmd. It's not a lone standing module so I also didn't add it to the table. But if we would like these items to be added, I can.

- [ ] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

<!-- IF YOUR PULL REQUEST IS A DATA RELEASE, PLEASE REMOVE THE [HTML COMMENT TAG](https://html.com/tags/comment-tag/) FROM THE SECTION BELOW AND COMPLETE THE CHECKLIST-->

jashapiro commented on analyses/snv-callers/scripts/03-calculate_tmb.R:
```diff
@@ -72,9 +72,15 @@ option_list <- list(
+    non-synonymous mutations, according to maftools definition.",
+    metavar = "character"
+  ),
+  make_option(
+    opt_str = "--nonsynfilter_focr", action = "store_true",
+    default = FALSE, help = "If TRUE, filter out synonymous mutations, keep
+    non-synonymous mutations, according to Friends of Cancer Research definition.",
     metavar = "character"
```
I like this addition!

jashapiro reviewed (commented):
Now that I have looked at this NB, I will confirm what I think I said in my review of #741, namely that I think these two notebooks can be combined, as this is kind of the crux of things at this point. 

If we start with the MAF file, we should be getting the same mutation counts by either script: any discrepancy (even off by 1!) indicates a difference in methodology and/or an error in one script or the other. At this point, we know that the differences are small in counts, and the question is whether the FOCR vs. maftools method is accounting for that difference. I suspect it is, though there may be some remaining edge cases that could result in some remaining differences. 

The second part (covered more in #741) is a difference in the definitions of the reference regions. I just looked back at that data and have another comment there...

cansavvy commented on analyses/snv-callers/explore_variant_classifications/explore_var_class_discrepancies.Rmd:
````diff
@@ -0,0 +1,193 @@
+
+Summarize by definition inclusion/exclusion. 
+
+```{r}
+summarized_df %>%
+  dplyr::group_by(var_def, dataset) %>%
+  dplyr::summarize(total_count = sum(counts)) %>%
+  ggplot2::ggplot(ggplot2::aes(
+    x = var_def,
+    y = total_count,
+    fill = var_def
+  )) +
+  ggplot2::theme_classic() +
+  ggplot2::geom_bar(
+    position = "dodge", stat = "identity"
+  ) +
+  ggplot2::theme(axis.text.x = ggplot2::element_text(angle = 45, hjust = 1)) +
+  ggplot2::xlab("") +
+  ggplot2::ylab("Number of TCGA mutations") +
+  colorblindr::scale_fill_OkabeIto() +
+  ggplot2::facet_wrap(~dataset)
+```
````
Which one do you like better?

cansavvy replied:
I like the summarized one so I deleted the other one.

cansavvy commented on analyses/snv-callers/explore_variant_classifications/explore_var_class_discrepancies.Rmd:
```diff
@@ -0,0 +1,193 @@
+  "..",
+  "results",
+  "consensus",
+  "pbta-snv-consensus-mutation.maf.tsv.gz"
```
I switched this for PBTA, but realized the TCGA tmb data is not in the data release, so I'll file an issue about that.

cansavvy replied:
https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/732#issuecomment-686586352

cansavvy commented on analyses/snv-callers/explore_variant_classifications/explore_var_class_discrepancies.Rmd:
````diff
@@ -0,0 +1,193 @@
+  ggplot2::facet_wrap(~dataset)
+```
+
+The discrepant classes don't seem to show up often. 
````
I just have it printing out the table. Currently this notebook doesn't compare to D3B, but I can add that in #741

cansavvy commented:
I'm not exactly sure if this addresses all your suggestions, but I think it does? I like keeping the notebook here and the one in #741 separate, since they are slightly different questions (even though they are related).

jashapiro reviewed (approved):
After our discussion, the comparison of focr and maftools results will go in #741, so this should be fine as is!
